src/HotChocolate/Data/test/Data.Filters.Tests/Convention/__snapshots__/FilterConventionScopeTests.FilterConvention_Should_Work_When_ConfiguredWithType.snap

```diff
@@ -7,50 +7,18 @@ type Foo {
 }
 
 type QueryType {
-  foos(where: FooFilter): [Foo]
-  foosBar(where: Bar_FooFilter): [Foo]
+  foos(where: ObjectFilterInput): [Foo]
```

Comment:
This doesnt look right

Reply:
@michaelstaib  The problem is in the UseFiltering extension
https://github.com/ChilliCream/hotchocolate/blob/d768752220a45e39a61d4beaf47c040edcb3eb31/src/HotChocolate/Data/src/Data/Filters/Extensions/FilterObjectFieldDescriptorExtensions.cs#L89-L90

`definition.ResultType` is of type `Object`.. I am not sure if this is wrong or if we have to work around this. 
We need to fix this before the merge though. Because this would mean that `[UseFiltering]` is in some cases wrong

Reply:
This was a bug ... I have fixed it now.

Reply:
Also we will now fail if we cannot infer a proper type:

```
if (definition.ResultType is null || 
                            definition.ResultType == typeof(object) ||
                            !c.TypeInspector.TryCreateTypeInfo(
                                definition.ResultType, out ITypeInfo? typeInfo))
                        {
                            throw new ArgumentException(
                                FilterObjectFieldDescriptorExtensions_UseFiltering_CannotHandleType,
                                nameof(descriptor));
                        }
```